src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -30,9 +31,10 @@ pub(super) struct EventProcessor<T: 'static> {
     pub(super) randr_event_offset: c_int,
     pub(super) devices: RefCell<HashMap<DeviceId, Device>>,
     pub(super) xi2ext: XExtension,
+    pub(super) xkbext: XExtension,
     pub(super) target: Rc<RootELW<T>>,
-    pub(super) mod_keymap: ModifierKeymap,
-    pub(super) device_mod_state: ModifierKeyState,
+    pub(super) kb_state: KbState,
```

Comment:
Do we need a fallback in case XKB isn't available, or can we just assume that XKB is present on modern machines? If not, I can make a future PR that adds a fallback.

Reply:
I don't think it's worth trying to support anything other than XKB. If that decision breaks things for enough people, we should perhaps reevaluate, but I don't expect that to be an issue.